obspy/core/utcdatetime.py

```diff
@@ -981,63 +987,63 @@ def __unicode__(self):
         """
         return str(self.__str__())
 
+    def _operate(self, other, op_func):
```

Comment:
I'm really not that deep into the fuzzy comparison problem but would it make sense to first convert anything to a `UTCDateTime` object and then just compare those? Might be easier to keep the logic straight in that fashion.

Reply:
or to have it maybe more lightweight using nanoseconds (```long``` datatype)?

Reply:
I modified the function to always use the nanoseconds comparison as suggested. Look ok to you guys?